### ISSUE TITLE
Deploy fails:   Error installing chef: rack requires Ruby version >= 2.2.2.

### ISSUE THREAD

**Issue by Artemmkin** [travis-ci/dpl#485]
It seems like deploy fails due to the fact that rack requires Ruby version >= 2.2.2.

```log
Fetching: dpl-1.8.18.travis.1530.3.gem (100%)
Successfully installed dpl-1.8.18.travis.1530.3
1 gem installed
dpl.1
Installing deploy dependencies
Fetching: plist-3.1.0.gem (100%)
Successfully installed plist-3.1.0
Fetching: method_source-0.8.2.gem (100%)
Successfully installed method_source-0.8.2
Fetching: slop-3.6.0.gem (100%)
Successfully installed slop-3.6.0
Fetching: coderay-1.1.1.gem (100%)
Successfully installed coderay-1.1.1
Fetching: pry-0.10.4.gem (100%)
Successfully installed pry-0.10.4
Fetching: rack-2.0.1.gem (100%)
ERROR:  Error installing chef:
	rack requires Ruby version >= 2.2.2.
/home/travis/.rvm/rubies/ruby-1.9.3-p551/lib/ruby/site_ruby/1.9.1/rubygems/core_ext/kernel_require.rb:54:in `require': cannot load such file -- chef (LoadError)
	from /home/travis/.rvm/rubies/ruby-1.9.3-p551/lib/ruby/site_ruby/1.9.1/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.18.travis.1530.3/lib/dpl/provider.rb:75:in `requires'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.18.travis.1530.3/lib/dpl/provider/chef_supermarket.rb:9:in `<class:ChefSupermarket>'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.18.travis.1530.3/lib/dpl/provider/chef_supermarket.rb:3:in `<class:Provider>'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.18.travis.1530.3/lib/dpl/provider/chef_supermarket.rb:2:in `<module:DPL>'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.18.travis.1530.3/lib/dpl/provider/chef_supermarket.rb:1:in `<top (required)>'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.18.travis.1530.3/lib/dpl/provider.rb:57:in `const_get'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.18.travis.1530.3/lib/dpl/provider.rb:57:in `block in new'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.18.travis.1530.3/lib/dpl/cli.rb:41:in `fold'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.18.travis.1530.3/lib/dpl/provider.rb:54:in `new'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.18.travis.1530.3/lib/dpl/cli.rb:31:in `run'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.18.travis.1530.3/lib/dpl/cli.rb:7:in `run'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.18.travis.1530.3/bin/dpl:5:in `<top (required)>'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/bin/dpl:23:in `load'
	from /home/travis/.rvm/gems/ruby-1.9.3-p551/bin/dpl:23:in `<main>'
failed to deploy
```

I think this PR #443 may fix this by pinning the version of rack in chef_supermarket provider. <br>Could consider merging that PR?

**Comment by serjs** [comment]
+1 For PR #443, rack 2.0.1 released on June 30, after release chef deploy is not working anymore

**Comment by Artemmkin** [comment]
Just tested the changes with my own fork and it works. Unless you merge #443 deploy to chef supermarket won't work.

**Comment by BanzaiMan** [comment]
Sorry for the delay. I've merged #443. Once https://travis-ci.org/travis-ci/dpl/jobs/149176953 passes and the resultant gem is uploaded, it will be available with:

```yaml
deploy:
  edge: true
  ⋮ #rest
```

Thanks.
